adding PHP documentation for exporters, resources, propagation (#2415)
open-telemetry/opentelemetry.io: content/en/docs/instrumentation/php/propagation.md

---
title: Propagation
description: Context propagation for the PHP API

> aliases: [/docs/instrumentation/php/api/propagation]

Reviewer: Drop the alias.

Reviewer: You dropped too many aliases. This one (in the new file) needed to be removed, not the one in `content/en/docs/instrumentation/php/getting-started.md`. Restore that alias.

weight: 7
---

Propagation is the mechanism that moves data between services and processes.
Although not limited to tracing, it is what allows traces to build causal
information about a system across services that are arbitrarily distributed
across process and network boundaries.

OpenTelemetry provides a text-based approach to propagate context to remote
services using the [W3C Trace Context](https://www.w3.org/TR/trace-context/)
HTTP headers.

## Context propagation with frameworks and libraries

Auto-instrumentation exists for some popular PHP frameworks (eg Symfony,
Laravel, Slim) and HTTP libraries propagate context for incoming and outgoing
HTTP requests.

> **It is highly recommend that you use auto-instrumentation or instrumentation

Reviewer: ```suggestion
**We highly recommend that you use auto-instrumentation or instrumentation
```

libraries to propagate context**. Although it is possible to propagate context
manually, the PHP auto-instrumentation and instrumentation libraries are
well-tested and easier to use.

### Incoming

Auto-instrumentation for frameworks which implement the
[PSR-15](https://www.php-fig.org/psr/psr-15/) `RequestHandlerInterface` will
automatically extract W3C tracecontext headers, create a root span, and set a
remote parent for the root span.

> ```shell
> $ composer require open-telemetry/opentelemetry-auto-psr15

Reviewer: ```suggestion
```shell
composer require open-telemetry/opentelemetry-auto-psr15
```
